src/components/functional/ledger/ledger-addresses/ledger-addresses.tsx

```diff
@@ -24,50 +26,120 @@ export class LedgerAddresses {
   }
 
   handleSelectAccount(accountDerivationIndex: number) {
-    return () => {
+    return (event: MouseEvent) => {
+      event.preventDefault();
       this.selectAccount.emit(accountDerivationIndex);
     };
   }
 
+  private handlePageChange(event: CustomEvent) {
+    event.preventDefault();
+    event.stopPropagation();
+    this.pageChange.emit(event.detail);
+  }
+
+  private processLedgerAddressIndex(accountDerivation: ILedgerAccount) {
+    return Number(accountDerivation.index + 1).toLocaleString();
+  }
+
   render() {
-    const isAddressesLoading = this.accountScreenData.isLoading || this.accountScreenData.accounts.length === 0;
+    const isPageChanging = this.accountScreenData.isLoading;
+    const isAddressesLoadingInitially = this.accountScreenData.accounts.length === 0;
     const totalPages = Math.ceil(TOTAL_ADDRESSES_COUNT / this.accountScreenData.addressesPerPage);
+    const isSelectedWalletOnPage = this.accountScreenData.accounts.some(accountDerivation => accountDerivation.index === this.selectedIndex);
+    const isAccessWalletDisabled = !isSelectedWalletOnPage && !isPageChanging;
+    const lastIndexOfPage = this.accountScreenData.startIndex + this.accountScreenData.addressesPerPage;
+    const isSingleDigitIndex = lastIndexOfPage <= 10;
+    const isIndexBelowOneHundred = !isSingleDigitIndex && lastIndexOfPage < 100;
+    const isIndexInTheHundreds = !isIndexBelowOneHundred && !isSingleDigitIndex && lastIndexOfPage < 1000;
+    const isIndexInTheThousands = lastIndexOfPage >= 1000;
+
+    const ledgerAddressesClasses: Record<string, string> = {
+      buttonTooltip: 'mvx:absolute mvx:top-0 mvx:h-12 mvx:left-0 mvx:right-0',
+      preloaderItem: 'mvx:h-16! mvx:border mvx:border-solid mvx:border-transparent mvx:rounded-lg! mvx:flex mvx:items-center mvx:w-full! mvx:p-4',
+      preloaderItemCheckbox: 'mvx:h-4! mvx:mr-2 mvx:min-w-4! mvx:w-4! mvx:rounded-full! mvx:bg-neutral-700!',
+      preloaderItemAddress: 'mvx:w-40! mvx:h-4! mvx:bg-neutral-700! mvx:rounded-lg! mvx:mr-auto',
+      preloaderItemBalance: 'mvx:w-24! mvx:h-4! mvx:bg-neutral-700! mvx:rounded-lg! mvx:ml-2',
+      preloaderItemIndex: classNames('mvx:mr-2 mvx:h-4! mvx:bg-neutral-700! mvx:rounded-lg!', {
```

Comment:
why not write it directly in TSX?

Reply:
This was supposedly written outside the component normally, for better readability, but I had to move it here because I had conditions depending on inner variables.